### PR TITLE
add air-gapped mode to trivy

### PR DESCRIPTION
**Pull request by developer-guy**
Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>

# Changes

I just tried to air-gapped support of Trivy for supporting air-gapped installations by enabling a flag named `AIR_GAPPED_ENABLED.`

> _https://aquasecurity.github.io/trivy/v0.36/docs/advanced/air-gap/_

/kind feature

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations][authoring]
- [x] Includes [docs][docs] (if user facing)
- [x] Includes [tests][tests] (for new tasks or changed functionality)
  - See the [end-to-end testing documentation][e2e] for guidance and CI details.
- [x] Meets the [Tekton contributor standards][contributor] (including functionality, content, code)
- [x] Commit messages follow [commit message best practices][commit]
- [x] Has a kind label. You can add one by adding a comment on this PR that
  contains `/kind <type>`. Valid types are bug, cleanup, design, documentation,
  feature, flake, misc, question, tep
- [ ] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md) for more details._

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]: https://github.com/tektoncd/catalog/issues/413
[authoring]: https://github.com/tektoncd/catalog/blob/main/recommendations.md
[docs]: https://github.com/tektoncd/community/blob/master/standards.md#docs
[tests]: https://github.com/tektoncd/community/blob/master/standards.md#tests
[e2e]: https://github.com/tektoncd/catalog/blob/main/CONTRIBUTING.md#end-to-end-testing
[contributor]: https://github.com/tektoncd/community/blob/main/standards.md
[commit]: https://github.com/tektoncd/community/blob/master/standards.md#commit-messages

### CODE REVIEW

File: task/trivy-scanner/0.1/tests/pvc.yaml [hunk 1]
```diff
@@ -8,5 +8,5 @@ spec:
     - ReadWriteOnce
   resources:
     requests:
-      storage: 100Mi
+      storage: 200Mi
```
Comment (vinamra28) [inline task/trivy-scanner/0.1/tests/pvc.yaml:11]: any reason for increasing the storage size? Not a blocker, just asking out of curiosity

File: task/trivy-scanner/0.1/trivy-scanner.yaml [hunk 2]
```diff
@@ -31,13 +31,28 @@ spec:
     - name: IMAGE_PATH
       description: Image or Path to be scanned by trivy.
       type: string
+    - name: AIR_GAPPED_ENABLED
+      description: a flag enabling Air-Gapped mode
+      type: string
+      default: "false"
```
Comment (vinamra28) [inline task/trivy-scanner/0.1/trivy-scanner.yaml:37]: this is a feature. I believe we should bump up the version 🙂

**Review by vinamra28 — COMMENTED** [review]
/ok-to-test

File: task/trivy-scanner/0.2/trivy-scanner.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: trivy-scanner
+  labels:
+    app.kubernetes.io/version: "0.1"
```
Comment (vinamra28) [inline task/trivy-scanner/0.2/trivy-scanner.yaml:7]:
```suggestion
    app.kubernetes.io/version: "0.2"
```

File: task/trivy-scanner/0.2/trivy-scanner.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      workingDir: $(workspaces.manifest-dir.path)
+      script: |
+        #!/usr/bin/env sh
+          cmd="trivy $* "
+          if [ "$(params.AIR_GAPPED_ENABLED)" = "true" ]; then
+            echo "Air-Gapped mode enabled"
+            TRIVY_TEMP_DIR=$(mktemp -d)
+            trivy --cache-dir $TRIVY_TEMP_DIR image --download-db-only
+            tar -cf ./db.tar.gz -C $TRIVY_TEMP_DIR/db metadata.json trivy.db
+            rm -rf $TRIVY_TEMP_DIR
```
Comment (vinamra28) [inline task/trivy-scanner/0.2/trivy-scanner.yaml:50]: I am suspecting that an organisation will have to enable traffic atleast for once in order to download the db, right?

Comment (developer-guy) [reply task/trivy-scanner/0.2/trivy-scanner.yaml:50]: Yep, this is not a totally network-free solution, I know you assumed that way, as we said, `air-gapped.`

**Review by vinamra28 — APPROVED** [review]

File: task/trivy-scanner/0.2/trivy-scanner.yaml (hunk 4 above)
Comment (keskad) [reply task/trivy-scanner/0.2/trivy-scanner.yaml:50]: Maybe is there a way to provide a PVC with already downloaded DB or to use a PROXY to download that DB?

Comment (vinamra28) [reply task/trivy-scanner/0.2/trivy-scanner.yaml:50]: @keskad, not sure about the proxy part but for the other part, ie, way to provide a PVC, yes, there can be 2 ways:
1. If size of downloaded db is less than 1MB then one can use ConfigMap
OR
2. Create a PVC, first mount it to a Pod and use `kubectl cp` command to copy the downloaded db from your local machine to pvc and then subsequently use that pvc wherever you want https://www.middlewareinventory.com/blog/kubectl-cp-example/

**Pull request merged**